services/121-service/src/notifications/message-incoming/message-incoming.service.ts

```diff
@@ -145,8 +145,12 @@ export class MessageIncomingService {
       callbackData.MessageStatus === TwilioStatus.delivered ||
       callbackData.MessageStatus === TwilioStatus.failed
     ) {
+      const sidCondition = callbackData.SmsSid
+        ? Equal(callbackData.SmsSid)
+        : undefined;
+
       const tryWhatsapp = await this.tryWhatsappRepository.findOne({
-        where: { sid: callbackData.SmsSid },
+        where: sidCondition ? { sid: sidCondition } : {},
         relations: ['registration'],
       });
```

Comment:
This doesn't look safe. I don't think it will return `null` if the object is empty (due to the typeofm bug). Instead, I think `this.tryWhatsappRepository.findOne` should be called conditionally based on whether `callbackData.SmsSid` is defined.